test_conformance/pipes/main.c

```diff
@@ -15,72 +15,94 @@
 //
 #include "harness/compat.h"
 
+#include "harness/testHarness.h"
+#include "procs.h"
 #include <stdio.h>
 #include <string.h>
-#include "procs.h"
-#include "harness/testHarness.h"
+
+test_status InitCL(cl_device_id device) {
+  auto version = get_device_cl_version(device);
+  if (version < Version(2, 0)) {
+    return TEST_SKIP;
+  }
+
+  int error;
+  cl_uint max_packet_size;
+  error = clGetDeviceInfo(device, CL_DEVICE_PIPE_MAX_PACKET_SIZE,
+                          sizeof(max_packet_size), &max_packet_size, NULL);
+  if (error != CL_SUCCESS) {
+    print_error(error, "Unable to get pipe max packet size");
+    return TEST_FAIL;
+  }
+
+  if (max_packet_size == 0) {
+    return TEST_SKIP;
```

Comment:
Hmm, just spotted another problematic case: a reported size of zero should fail the test on 2.0-2.2 implementations, not skip it. I suggest adding ` && (version > Version(2,2))` to the condition.